bixlight/src/main/java/com/github/shingyx/bixlight/BixlightService.java

```diff
@@ -39,12 +42,18 @@ public void onAccessibilityEvent(AccessibilityEvent event) {
                 "onAccessibilityEvent: [type] %s [time] %s [activeWindowPackage] %s",
                 AccessibilityEvent.eventTypeToString(event.getEventType()), event.getEventTime(), activeWindowPackage));
 
-        if (!BIXBY_PACKAGE.equals(activeWindowPackage)) {
+        long currentMillis = Calendar.getInstance().getTimeInMillis();
+        boolean runTooSoon = (currentMillis - lastRunMillis) < MAX_RUN_FREQUENCY_MS;
+
+        if (runTooSoon || !BIXBY_PACKAGE.equals(activeWindowPackage)) {
             return;
         }
 
         if (setupCameraIfNeeded()) {
             try {
+                String action = torchEnabled ? "OFF" : "ON";
+                Log.v(TAG, "turning " + action + " torch");
+                lastRunMillis = currentMillis;
```

Comment:
I prefer having the minimum number of lines in the `try` block to make it really obvious what could throw the exception.